transmogrifier/sources/xml/datacite.py

```diff
@@ -54,7 +55,8 @@ def get_optional_fields(self, xml: Tag) -> dict | None:
                 if self.valid_content_types([content_type]):
                     fields["content_type"] = [content_type]
                 else:
-                    return None
+                    message = f'Record skipped based on content type: "{content_type}"'
+                    raise SkippedRecordEvent(source_record_id, message)
```

Comment:
See comments above about possible reworking of these custom exceptions.

Even if we don't go that route, I think perhaps the `message` should be the first argument, and positional, and then `source_record_id` be a named argument is optional.